interfaces/connection_test.go

```diff
@@ -0,0 +1,189 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package interfaces
+
+import (
+	. "gopkg.in/check.v1"
+	"reflect"
+
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/snap/snaptest"
+)
+
+type connSuite struct {
+	plug *snap.PlugInfo
+	slot *snap.SlotInfo
+}
+
+var _ = Suite(&connSuite{})
+
+func (s *connSuite) SetUpTest(c *C) {
+	consumer := snaptest.MockInfo(c, `
+name: consumer
+apps:
+    app:
+plugs:
+    plug:
+        interface: interface
+        attr: value
+`, nil)
+	s.plug = consumer.Plugs["plug"]
+	producer := snaptest.MockInfo(c, `
+name: producer
+apps:
+    app:
+slots:
+    slot:
+        interface: interface
+        attr: value
+`, nil)
+	s.slot = producer.Slots["slot"]
+}
+
+func (s *connSuite) TestStaticSlotAttrs(c *C) {
+	slot := NewConnectedSlot(s.slot, nil)
+	c.Assert(slot, NotNil)
+
+	_, err := slot.StaticAttr("unknown")
+	c.Assert(err, NotNil)
+	c.Assert(err, ErrorMatches, `attribute "unknown" not found`)
+
+	attrs := slot.StaticAttrs()
+	c.Assert(attrs, DeepEquals, map[string]interface{}{
+		"attr": "value",
+	})
+}
+
+func (s *connSuite) TestStaticPlugAttrs(c *C) {
+	plug := NewConnectedPlug(s.plug, nil)
+	c.Assert(plug, NotNil)
+
+	_, err := plug.StaticAttr("unknown")
+	c.Assert(err, NotNil)
+	c.Assert(err, ErrorMatches, `attribute "unknown" not found`)
+
+	attrs := plug.StaticAttrs()
+	c.Assert(attrs, DeepEquals, map[string]interface{}{
+		"attr": "value",
+	})
+}
+
+func (s *connSuite) TestDynamicSlotAttrs(c *C) {
+	attrs := map[string]interface{}{
+		"foo":    "bar",
+		"number": int(100),
+	}
+	slot := NewConnectedSlot(s.slot, attrs)
+	c.Assert(slot, NotNil)
+
+	val, err := slot.Attr("foo")
+	c.Assert(err, IsNil)
+	c.Assert(val, Equals, "bar")
+
+	val, err = slot.Attr("attr")
+	c.Assert(err, IsNil)
+	c.Assert(val, Equals, "value")
+
+	val, err = slot.Attr("number")
+	c.Assert(err, IsNil)
+	c.Assert(reflect.TypeOf(val).Kind(), Equals, reflect.Int64)
```

Comment:
For the record, this check is unnecessary given the following line. Equals takes the type into account as well. Same below.